colossalai/tensor/colo_parameter.py

```diff
@@ -1,28 +1,35 @@
 from .colo_tensor import ColoTensor
 from .const import TensorType
 import torch
+from colossalai.tensor import TensorSpec, dist_spec
+from copy import copy
 
 
 class ColoParameter(ColoTensor):
     r"""A kind of ColoTensor to be considered as a module parameter.
 
     """
 
-    def __init__(self, *args, **kargs):
-        super().__init__(*args, **kargs)
-        self._type = TensorType.MODEL
+    def __new__(cls,
+                data: torch.Tensor,
+                requires_grad: bool = True,
+                spec: TensorSpec = TensorSpec(dist_spec.replicate())) -> 'ColoParameter':
+        if data is None:
+            data = torch.empty(0)
+        return torch.Tensor._make_subclass(cls, data, requires_grad)
```

Comment:
Even we inherit from ColoTensor, we still need rewrite a new() from torch.Tensor._make_subclass? May I know the reason?

Reply:
Their init function takes different arguments. Generally, `__new__` and `__init__` should have the same function signature (except the first argument). Besides, `requires_grad` is set in `__new__`. ColoParameter receives `requires_grad`, but ColoTensor doesn't.

Reply:
So, torch.Tensor._make_subclass is the same as torch.Tensor.__new__ in this functionality?

Reply:
I think so.

Reply:
Remeber we have an attr in ColoTendor
` self._type = TensorType.NONMODEL``
you should also set it to TensorType.MODEL in new!

Reply:
They are set in `__init__`